### PR TITLE
mgr: influx: catch KeyError when looking up pool name

### PR DESCRIPTION
**Pull request by tobias-urdin**
It can lookup the pool name when a new pool is created
and the pool_info doesn't contain it causing a KeyError.

Fixes bug https://tracker.ceph.com/issues/56130

Signed-off-by: Tobias Urdin <tobias.urdin@binero.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [X] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/influx/module.py [hunk 1]
```diff
@@ -233,11 +233,15 @@ def get_pg_summary_osd(self, pool_info: Dict[str, str], now: str) -> Iterator[Di
     def get_pg_summary_pool(self, pool_info: Dict[str, str], now: str) -> Iterator[Dict[str, Any]]:
         pool_sum = self.get('pg_summary')['by_pool']
         for pool_id, stats in pool_sum.items():
+            try:
+                pool_name = pool_info[pool_id]
+            except KeyError:
+                continue
```
Comment (ljflores) [inline src/pybind/mgr/influx/module.py:239]: Let's log the exception. Something like: 
```suggestion
            try:
                pool_name = pool_info[pool_id]
            except Exception as e:
                self.log.error('Unable to find pool name for pool {}: {}'.format(pool_id, e))
                continue
```

**Comment by ljflores** [comment]
jenkins retest this please

File: src/pybind/mgr/influx/module.py (hunk 1 above)
Comment (tobias-urdin) [reply src/pybind/mgr/influx/module.py]: Thanks!

File: src/pybind/mgr/influx/module.py [hunk 2]
```diff
@@ -233,11 +233,16 @@ def get_pg_summary_osd(self, pool_info: Dict[str, str], now: str) -> Iterator[Di
     def get_pg_summary_pool(self, pool_info: Dict[str, str], now: str) -> Iterator[Dict[str, Any]]:
         pool_sum = self.get('pg_summary')['by_pool']
         for pool_id, stats in pool_sum.items():
+            try:
+                pool_name = pool_info[pool_id]
+            except Exception as e:
```
[the remark this answers is not in the document]
Comment (ljflores) [reply src/pybind/mgr/influx/module.py:238]: Makes sense. I still think we should keep the error log if this change is made though so we know which pool it failed on.

[the remark this answers is not in the document]
Comment (tobias-urdin) [reply src/pybind/mgr/influx/module.py]: ack, updated according to above.

**Comment by kamoltat** [comment]
0 related failures

### **yuriw-2022-06-23_14:17:25-rados-wip-yuri6-testing-2022-06-22-1419-distro-default-smithi**

jobid: [6894622, 6894626, 6894629, 6894631]
description: rados/singleton-bluestore/{all/cephtool mon_election/classic msgr-failures/many msgr/async-v2only objectstore/bluestore-comp-lz4 rados supported-random-distro$/{rhel_8}}
failure_reason: Command failed (workunit test cephtool/test.sh) on smithi189 with status 1: 'mkdir -p -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && cd -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && CEPH_CLI_TEST_DUP_COMMAND=1 CEPH_REF=fad4b1c200ee6a758bd948f031903dd98c630b4c TESTDIR="/home/ubuntu/cephtest" CEPH_ARGS="--cluster ceph" CEPH_ID="0" PATH=$PATH:/usr/sbin CEPH_BASE=/home/ubuntu/cephtest/clone.client.0 CEPH_ROOT=/home/ubuntu/cephtest/clone.client.0 CEPH_MNT=/home/ubuntu/cephtest/mnt.0 adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 3h /home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh'
traceback: check_response erasure-code didn't find erasure-code in output
tracker: NA
created_tracker: https://tracker.ceph.com/issues/56384

jobid: [6894623, 6894627, 6894630, 6894632]
description: rados/rook/smoke/{0-distro/ubuntu_20.04 0-kubeadm 0-nvme-loop 1-rook 2-workload/radosbench cluster/1-node k8s/1.21 net/host rook/master}
failure_reason: 'wait for operator' reached maximum tries (90) after waiting for 900 seconds
traceback: 
tracker: https://tracker.ceph.com/issues/52321
created_tracker: 

jobid: [6894628, 6894633, 6894631]
description: rados/upgrade/parallel/{0-random-distro$/{centos_8.stream_container_tools_crun} 0-start 1-tasks mon_election/classic upgrade-sequence workload/{ec-rados-default rados_api rados_loadgenbig rbd_import_export test_rbd_api test_rbd_python}}
failure_reason: Command failed (workunit test cls/test_cls_rgw.sh) on smithi162 with status 1: 'mkdir -p -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && cd -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && CEPH_CLI_TEST_DUP_COMMAND=1 CEPH_REF=pacific TESTDIR="/home/ubuntu/cephtest" CEPH_ARGS="--cluster ceph" CEPH_ID="0" PATH=$PATH:/usr/sbin CEPH_BASE=/home/ubuntu/cephtest/clone.client.0 CEPH_ROOT=/home/ubuntu/cephtest/clone.client.0 CEPH_MNT=/home/ubuntu/cephtest/mnt.0 adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 3h /home/ubuntu/cephtest/clone.client.0/qa/workunits/cls/test_cls_rgw.sh'
traceback:  2022-06-23T14:59:37.981 INFO:tasks.workunit.client.0.smithi162.stdout:[  FAILED  ] cls_rgw.index_suggest_complete
2022-06-23T14:59:37.981 INFO:tasks.workunit.client.0.smithi162.stdout:[  FAILED  ] cls_rgw.index_list
2022-06-23T14:59:37.981 INFO:tasks.workunit.client.0.smithi162.stdout:[  FAILED  ] cls_rgw.index_list_delimited
tracker: https://tracker.ceph.com/issues/55789, https://tracker.ceph.com/issues/55853
created_tracker:

DEAD job

jobid: 6894621
description: NA
failure_reason: '082ae7ef4302fa54665ed0a2535e8e254118dcfd' not found in repo: git://git.ceph.com/git/teuthology.git!
traceback:
tracker:
created_tracker:


### **yuriw-2022-06-23_03:08:39-rados-wip-yuri6-testing-2022-06-22-1419-distro-default-smithi**

jobid: [6893579, 6893654, 6893734, 6893811]
description: rados/singleton-bluestore/{all/cephtool mon_election/classic msgr-failures/many msgr/async-v2only objectstore/bluestore-comp-lz4 rados supported-random-distro$/{rhel_8}}
failure_reason: Command failed (workunit test cephtool/test.sh) on smithi064 with status 1: 'mkdir -p -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && cd -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && CEPH_CLI_TEST_DUP_COMMAND=1 CEPH_REF=fad4b1c200ee6a758bd948f031903dd98c630b4c TESTDIR="/home/ubuntu/cephtest" CEPH_ARGS="--cluster ceph" CEPH_ID="0" PATH=$PATH:/usr/sbin CEPH_BASE=/home/ubuntu/cephtest/clone.client.0 CEPH_ROOT=/home/ubuntu/cephtest/clone.client.0 CEPH_MNT=/home/ubuntu/cephtest/mnt.0 adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 3h /home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh'
traceback:
tracker: https://tracker.ceph.com/issues/56384
created_tracker:

jobid: [6893583, 6893664, 6893743, 6893822, 6893822]
description: rados/rook/smoke/{0-distro/ubuntu_20.04 0-kubeadm 0-nvme-loop 1-rook 2-workload/radosbench cluster/1-node k8s/1.21 net/host rook/master}
failure_reason: 'wait for operator' reached maximum tries (90) after waiting for 900 seconds
traceback:
tracker: https://tracker.ceph.com/issues/52321
created_tracker:

jobid: [6893667, 6893824]
description: rados/upgrade/parallel/{0-random-distro$/{centos_8.stream_container_tools_crun} 0-start 1-tasks mon_election/classic upgrade-sequence workload/{ec-rados-default rados_api rados_loadgenbig rbd_import_export test_rbd_api test_rbd_python}}
failure_reason: Command failed (workunit test cls/test_cls_rgw.sh) on smithi088 with status 1: 'mkdir -p -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && cd -- /home/ubuntu/cephtest/mnt.0/client.0/tmp && CEPH_CLI_TEST_DUP_COMMAND=1 CEPH_REF=pacific TESTDIR="/home/ubuntu/cephtest" CEPH_ARGS="--cluster ceph" CEPH_ID="0" PATH=$PATH:/usr/sbin CEPH_BASE=/home/ubuntu/cephtest/clone.client.0 CEPH_ROOT=/home/ubuntu/cephtest/clone.client.0 CEPH_MNT=/home/ubuntu/cephtest/mnt.0 adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 3h /home/ubuntu/cephtest/clone.client.0/qa/workunits/cls/test_cls_rgw.sh'
traceback:
tracker: https://tracker.ceph.com/issues/52321
created_tracker:

jobid: [6893596]
description: rados/mgr/{clusters/{2-node-mgr} debug/mgr mgr_ttl_cache/enable mon_election/classic random-objectstore$/{bluestore-comp-zlib} supported-random-distro$/{centos_8} tasks/prometheus}
failure_reason: "2022-06-23T03:54:30.960636+0000 mds.a (mds.0) 1 : cluster [WRN] evicting unresponsive client smithi165:z (4774), after 300.189 seconds" in cluster log
traceback:
tracker: https://tracker.ceph.com/issues/52876
created_tracker:

jobid: [6893630]
description: rados/cephadm/workunits/{agent/on mon_election/connectivity task/test_nfs}
failure_reason: Test failure: test_create_delete_cluster_idempotency (tasks.cephfs.test_nfs.TestNFS)
traceback:  cephadm exited with an error code: 1, stderr: ERROR: Daemon not found: mds.a.smithi059.tgjbmj. See `cephadm ls`
tracker: https://tracker.ceph.com/issues/56000
created_tracker:

**Pull request merged**